### PR TITLE
Modulereview

### PR DESCRIPTION
**Pull request by Jesse132**
Cloudinary upload and image return functionality completed. Various packets were installed to allow it to work. Various bugs such as form not validating required input and warnings have been fixed.

### CODE REVIEW

**Review by codingavatar — APPROVED** [review]
Looks good! How much of the server/cloudinary.js file works? A lot of lines are commented out.

**Comment by Jesse132** [comment]
> 

All the unnecessary  functions are commented out! The server file would load up all available files in my cloudinary account and show you the details of each photo - which aren't helpful. I've edited the getImage function to make it a two-in-one function where it uploads the image to cloudinary account and returns the cloudinary link to the uploaded image.

File: server/controllers.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ let putHelpfulReview = (review_id) => {
 // eslint-disable-next-line camelcase
 let postReview = (product_id, rating, summary, body, name, email, photos, recommended, characteristics) => {
   // eslint-disable-next-line camelcase
-  return (client.post(`/reviews/${product_id}/${rating}/${summary}/${body}/${name}/${email}/${photos}/${recommended}/${characteristics}`));
+  return (client.post(`/reviews/?product_id=${product_id}&rating=${rating}&summary=${summary}&body=${body}&recommend=${recommended}&name=${name}&email=${email}&photos=${photos}&characteristics=${characteristics}`));
```
Comment (poioper) [inline server/controllers.js:56]: Maybe put the parameters like this? https://youtu.be/7-zLcsfR1q0?t=191

File: webpack.config.js [hunk 2]
```diff
@@ -24,6 +24,9 @@ module.exports = {
   resolve: {
     extensions: ['.js', '.jsx']
   },
+  node: {
+    fs: 'empty'
```
Comment (poioper) [inline webpack.config.js:28]: May I ask what is this for?

**Review by poioper — APPROVED** [review]
Great work! All looks good!

**Pull request merged**